keras/layers/convolutional.py

```diff
@@ -13,6 +13,37 @@
     from theano.sandbox.cuda import dnn
 
 
+def conv_output_length(input_length, filter_size, border_mode, stride):
+    assert border_mode in {'same', 'full', 'valid'}
+    if border_mode == 'same':
+        output_length = input_length
+    elif border_mode == 'full':
+        output_length = input_length + filter_size - 1
+    elif border_mode == 'valid':
+        output_length = input_length - filter_size + 1
+    return (output_length + stride - 1) // stride
+
+
+def pool_output_length(input_length, pool_size, ignore_border, stride):
+    if ignore_border:
+        output_length = input_length - pool_size + 1
+        output_length = (output_length + stride - 1) // stride
+    else:
+        if pool_size == input_length:
+            output_length = min(input_length, stride - stride % 2)
```

Comment:
`stride - stride % 2`? I find the computations in this function hard to follow in general. Justificative comments would be helpful.